_components/code-component.md

```diff
@@ -0,0 +1,221 @@
+*   `require`: - [Module require](https://nodejs.org/dist/latest-v5.x/docs/api/modules.html),
+*   `setTimeout`: - more on [setTimeout](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `clearTimeout`: - more on [clearTimeout](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `setInterval`: - more on [setInterval](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `clearInterval`: - more on [clearInterval](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `msg`: - Incoming message containing the payload from the previous step,
+*   `exports`: {},
+*   `messages`: - [Utility for convenient message creation](https://github.com/elasticio/elasticio-node/blob/master/lib/messages.js),
+*   `request`: - [Http Client](https://github.com/id0x3d/co-request),
+*   `wait`: - wait,
+*   `emitter`: user to emit messages and errors
+
+
+## Known limitations
+
+Only the code written in `Node.js` is supported at this moment.
+
+## Code component usage Examples
+
+Here are several examples where it's possible to see how can Code component be
+used. For purposes of displaying the functionality, we will use the
+{{site.data.tenant.name}} [Rest API]({{site.data.tenant.apiBaseUri}}/docs/#introduction).
+
+In the examples below, we make an HTTP request to retrieve your user information
+from our API and pass this information to the next stage of integration. We use
+Environment variables to retrieve your API username and API Key which is already
+defined in your account while logged into the platform Dashboard.
+
+The information that we are accessing is normally given back in the
+[JSON form like]({{site.data.tenant.apiBaseUri}}/docs/#users):
+
+```js
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "id": "54f4be3fe7d5224f91000001",
+  "first\_name": "John",
+  "last\_name": "Doe",
+  "email": "test@example.com"
+}
+```
+
+### Using Callbacks
+
+The simplest way to retrieve this information could be done using JavaScript callbacks.
+
+```js
+console.log('Starting Node.js code execution');
+
+var opts = {
+    uri: '{{site.data.tenant.apiBaseUri}}/v2/users',
+    auth: {
+        user: process.env.ELASTICIO\_API\_USERNAME,
+        pass: process.env.ELASTICIO\_API\_KEY
+    },
+    json: true
+};
+
+request.get(opts, function(err, response, body){
+    if (err) {
+        console.log(err);
+        return emitter.emit(err);
+    }
+
+    if (response.statusCode != 200) {
+        return emitter.emit(new Error("API responded with", response.statusCode));
+    }
+
+    // create message to emit
+    var data = messages.newMessageWithBody({
+        fullName: body.first\_name + " " + body.last\_name,
+        email: body.email,
+        userID: body.id
+    });
+
+    // Emit the data event
+    emitter.emit('data', data);
+
+    console.log('Message emitted');
+});
+```
+
+For a simple use case, this is sufficient, however, if we then decide to include
```

Comment:
My suggestion is to remove this paragraph.  I don't think it is useful to transition from explaination of component to how to code JS tutorial here.